lib/elixir/lib/task.ex

```diff
@@ -752,18 +789,29 @@ defmodule Task do
   a response waiting in the message queue this function will return
   `{:exit, :noproc}` as the result or exit reason can not be determined.
   """
-  @spec shutdown(t, timeout | :brutal_kill) :: {:ok, term} | {:exit, term} | nil
-  def shutdown(task, shutdown \\ 5000)
+  @spec shutdown(t, timeout :: non_neg_integer | :brutal_kill) ::
+          {:ok, term} | {:exit, term} | nil
+  def shutdown(task, timeout_or_brutal_kill \\ 5000)
 
-  def shutdown(%Task{pid: nil} = task, _) do
-    raise ArgumentError, "task #{inspect(task)} does not have an associated task process"
-  end
+  def shutdown(%Task{pid: pid, owner: owner} = task, timeout)
+      when is_timeout(timeout) or timeout == :brutal_kill do
+    cond do
+      timeout == :infinity ->
```

Comment:
`:infinity` is definitely a valid value here. `:infinity` is a valid shutdown value for supervisors and therefore it should be a valid value here too. Usually when we say timeout, `:infinity` is included, that's why we don't say it explicitly in the docs.

Reply:
Let's please revert the changes to `shutdown` function. Let's keep only the improve `timeout_or_brutal_kill` variable name for the signature and make sure to use `is_timeout(timeout)` in the last clause of the previous implementation.

Reply:
Actually, `shutdown` was a perfectly valid name too. Let's keep it.

Reply:
Besides `:infinity` being a valid timeout, what is the point of allowing that in the `shutdown` function.\
It will never shutdown, or am I missing something,
Otherwise I think we should raise and inform the user that is going to do something that doesn't make sense.